app/controllers/mixins/ems_common_angular.rb

```diff
@@ -287,6 +293,10 @@ def ems_form_fields
 
     private ############################
 
+    def metrics_default_database_name
+      ManageIQ::Providers::Redhat::InfraManager.history_database_name_for('4_0')
```

Comment:
can you test if `@ems` is a `ManageIQ::Providers::Redhat::InfraManager` and otherwise return '' ?
also make sure, you dont use explicit constants, rather use strings and do a safe_constantize

so
```ruby
if @ems.class.name == 'ManageIQ::Providers::Redhat::InfraManager'
  ManageIQ::Providers::Redhat::InfraManager.history_database_name_for('4.0')
end
```